### PR TITLE
api: add NetworkFenceClass CRD

### PR DESCRIPTION
**Pull request by Madhu-1**
Adding NetworkFenceClass which is going to be used to get the client details to be fenced.

The controller will watch for the NetworkFenceClass CR and sends a periodic request to all the sidecars registered with the matching driver name that registered the GET_CLIENTS_TO_FENCE caps to get the client details, once the controller gets the client details its will update the CSIAddonsNode object status with the client information that can be used to do Network fencing and unfencing.

### CODE REVIEW

**Review by nixpanic — COMMENTED** [review]
Consider adding a description to the documentation so that it becomes easier to understand what this is about.

File: api/csiaddons/v1alpha1/networkfenceclass_types.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022 The Kubernetes-CSI-Addons Authors.
```
[the remark this answers is not in the document]
Comment (Madhu-1) [reply api/csiaddons/v1alpha1/networkfenceclass_types.go:2]: :D this is auto-generated one

**Review by black-dragon74 — APPROVED** [review]
LGTM, thank you!

File: api/csiaddons/v1alpha1/networkfenceclass_types.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (nixpanic) [reply api/csiaddons/v1alpha1/networkfenceclass_types.go:2]: yes, it comes from `hack/boilerplate.go.txt`. Do you want to update that?

**Review by nixpanic — COMMENTED** [review]
LGTM, let me know if you want to correct the year in the header.

**Comment by Madhu-1** [comment]
> LGTM, let me know if you want to correct the year in the header.

@nixpanic will take care of in followup PR

**Review by nixpanic — APPROVED** [review]

**Pull request merged**